### DISCUSSION TITLE
NFCe sem destinatário

### DISCUSSION THREAD

**Discussion by grammobrasil** [TadaSoftware/PyNFe#150] [General]
Pessoal, boa tarde. Estou testando o PyNFE e para vendas ao consumidor (uma tele-entrega) e me ocorreu se é possível emitir uma NFC-e sem destinatário? Ao tentar omitir os dados o programa alerta que não seria possível, mas "legalmente" falando é possível emitir NFCe sem destinatário para vendas de até R$10.000,00, que é o que ocorre de fato em larga escala com a maioria das vendas do varejo.

Alguém tem uma sugestão sobre como proceder? Já tiveram esta situação? Obrigado!

**Comment by juniortada** [comment]
@grammobrasil "legalmente" não é que você possa emitir uma nota sem consumidor, você pode emitir sem identifica-lo.
Resumindo, você faz uma nota normal e no objeto cliente(destinatário) coloca apenas informações fiscais ou só cpf ou nome e cpf, etc. mas campos como indicador_ie=9 continuam sendo obrigatórios.
O nível de informações fiscais obrigatórios mesmo para destinatários não identificados varia de UF autorizador.
Exemplo UR PR:

```python
cliente = Cliente(indicador_ie=9)
nota_fiscal.cliente = cliente
```

**Reply by grammobrasil** [reply]
Entendi. Mas no fim estávamos estudando algumas coisas e vamos acabar optando por colocar os dados do destinatário. De qualquer forma valeu pela atenção, e parabéns, o app é muito bom!

**Reply by grammobrasil** [reply]
Boa noite, novamente com dúvida sobre a identificação do destinatário.
Estava lendo a legislação do RS é encontrei isso:

"NOTA - Fica dispensada a inclusão do nome e do CPF na NFC-e que documentar operações de valor inferior a R$ 200,00 (duzentos reais), exceto na hipótese em que o consumidor queira informá-los."
http://www.legislacao.sefaz.rs.gov.br/Site/Document.aspx?inpKey=271140

Será que existiria a possibilidade de omitir o cliente no PyNFE, em modelo parecido com cupom fiscal?

> @grammobrasil "legalmente" não é que você possa emitir uma nota sem consumidor, você pode emitir sem identifica-lo. Resumindo, você faz uma nota normal e no objeto cliente(destinatário) coloca apenas informações fiscais ou só cpf ou nome e cpf, etc. mas campos como indicador_ie=9 continuam sendo obrigatórios. O nível de informações fiscais obrigatórios mesmo para destinatários não identificados varia de UF autorizador. Exemplo UR PR:
> 
> ```python
> cliente = Cliente(indicador_ie=9)
> nota_fiscal.cliente = cliente
> ```

**Reply by juniortada** [reply]
@grammobrasil pelo que entendi sua dúvida continua a mesma, a legislação é a mesma com pequenas variações entre UF's, como mencionei anteriormente você pode omitir o nome e/ou CPF/CNPJ do cliente na emissão de NFC-e, mas ainda é obrigatório algumas informações fiscais.

**Reply by grammobrasil** [reply]
Somos uma empresa de delivery e de fato a legislação é um pouco confusa nesse caso (encontramos normas sugerindo obrigatoriedade dos dados do cliente e outras liberando). Mas no nosso caso nos parece que exigir CPF de um comprador de refeições fica um pouco desconfortável, e gostaríamos de emitir NFCe nos mesmos moldes de cupom fiscal ou como a SAT Fiscal de São Paulo.

Na verdade gostaria saber se consigo retirar a obrigatoriedade da inserção de dados do cliente sem comprometer a funcionalidade do pyNFE, que é excelente. Conseguiriam me dar alguma sugestão da melhor maneira de ajustar isso no app?

Recomendaria

**Reply by juniortada** [reply]
@grammobrasil desculpa, mas ainda não entendi a sua dúvida. 
Diariamente milhares de notas são emitidas sem informar nome ou cpf (ou ambos) utilizando o pynfe. Basta não informar.
O "cliente" da lib é apenas um objeto python que é instanciado para configuração de alguns dados fiscais que não dependem da identificação ou não do seu cliente. Não é porque o objeto python é instanciado que ele precisa contar as informações completas.
As informações são de responsabilidade do desenvolvedor que está utilizando a lib.

**Reply by grammobrasil** [reply]
Eu estava fazendo uma barbeiragem na verdade. Estava tentando chamar o objeto Cliente() deixando campos vazios ou não identificados, mas chamando ele dentro de Notafiscal(). Ao não chamar, consegui emitir de imediato.
Desculpem a função, sou relativamente novato no tema, mas MUITO obrigado pela atenção!

**Reply by juniortada** [reply]
Você até pode fazer dessa forma, só não recomendo.
```python
  nota_fiscal = NotasFiscal(
    cliente=Cliente(indicador_ie=9),
    emitente=Emitente(razao_socia="bla bla")  
  )
```

Aqui explico melhor:
https://github.com/TadaSoftware/PyNFe/issues/126

**Reply by grammobrasil** [reply]
Excelente. Mais uma vez, muito obrigado!

**Comment by leogregianin** [comment]
@juniortada, no schema `leiauteNFe_v4.00.xsd` o grupo `dest` tem ocorrência mínima igual a zero (`<xs:element name="dest" minOccurs="0">`, ou seja, é um grupo opcional no XML seguindo algumas regras, claro.

As regras que levantei foram: a omissão do grupo `destinatario` pode ser feita somente para NFC-e (modelo 65), que o cliente não seja estrangeira e que atenda valores limites específicos de cada UF.

Exemplo: https://atendimento.tecnospeed.com.br/hc/pt-br/articles/360015691134

**Reply by juniortada** [reply]
@leogregianin eu tive esse problema com a SEFAZ PR, entrei em contanto com o suporte deles e resposta que obtive foi a mesma que falei para o @grammobrasil, posso omitir dados como nome e cpf mas não outras informações obrigatórias. 
Inclusive nos primeiros dias existia um bug onde não estava autorizando notas com apenas cpf, reportei e foi corrigido.
Já essa obrigatoriedade do elemento dest não. Mas isso já tem alguns anos, podem ter corrigido.

**Comment by leogregianin** [comment]
Atualizando a discussão, conforme [esse teste](https://github.com/TadaSoftware/PyNFe/blob/main/tests/test_nfce_serializacao_sem_cliente.py), é possível [omitir totalmente a classe Cliente](https://github.com/TadaSoftware/PyNFe/blob/2f12e171ccb35ff38dd8c106207704e7c8cb131b/tests/test_nfce_serializacao_sem_cliente.py#L77) e com isso não gerará a tag do grupo `dest`.

**Reply by leogregianin** [reply]
@grammobrasil, para melhor organização das discussões do projeto, pode marcar essa discussão por respondida?
